File: js/find_stray_number.js
function stray(numbers) {
//   return 0;
  // loop through arr, and return num that !== current num

  let strayNum = numbers[0]

  if (strayNum !== numbers[1]) {
    if (numbers[1] === numbers[2]) {
      return strayNum
    }
  }

  for (let i = 0; i < numbers.length; i++) {
    if (numbers[i] !== strayNum) {
      strayNum = numbers[i]
      return strayNum
    }
  }
}

File: java/find_stray_number.java
import java.util.Arrays;

class Solution {
  static int stray(int[] numbers) {
    // set strayNum to first item in arr
    int strayNum = numbers[0];

    // edge case
    // if first two nums in arr are different
    if (strayNum != numbers[1]) {
    // and if second num equals third num
      if (numbers[1] == numbers[2]) {
        // return first num in arr
        return strayNum;
      }
    }

    // loop through arr
    for (int i = 0; i < numbers.length; i++) {
      // current num in arr is not equal to strayNum
      if (numbers[i] != strayNum) {
        // strayNum is now current num
        strayNum = numbers[i];
      return strayNum;
    }
  }

    return strayNum;
  }
}
